Civilian Population wont install

Background
----------
**CKAN Version:**
v1.22.6+e8e10ae2e98e

**KSP Version:**
1.3.1

**Operating System:**
Ubuntu 17.10 64bit

**Have you made any manual changes to your GameData folder (i.e., not via CKAN)?** 
Clean install

Problem
-------

Civilian Population wont install.

**What steps did you take in CKAN?**
mono ckan.exe install  CivilianPopulation

**What did you expect to happen?**
Install Civilian Population.

**What happened instead?**
Crash report

**CKAN error codes (if applicable):** 
```
mono ckan.exe install  CivilianPopulation
About to install...

 * Civilian Population 2.0.12 (github.com)

Continue? [Y/n] 

Downloading "https://github.com/rleroy/CivilianPopulation/releases/download/2.0.12/ksp-civilian-population-mod-2.0.12.zip"
0 kbps - downloading - 0 MB left - 100%             
Installing mod "CivilianPopulation 2.0.12"
Unhandled Exception:
CKAN.FileNotFoundKraken: Trying to install CivilianPopulation 2.0.12, but it's not downloaded or download is corrupted
  at CKAN.ModuleInstaller.Install (CKAN.CkanModule module, System.String filename) [0x00073] in <fce8ecce013a4e11ba75e87f5f8ce9eb>:0 
  at CKAN.ModuleInstaller.InstallList (System.Collections.Generic.ICollection`1[T] modules, CKAN.RelationshipResolverOptions options, CKAN.IDownloader downloader) [0x00194] in <fce8ecce013a4e11ba75e87f5f8ce9eb>:0 
  at CKAN.ModuleInstaller.InstallList (System.Collections.Generic.List`1[T] modules, CKAN.RelationshipResolverOptions options, CKAN.IDownloader downloader) [0x00028] in <fce8ecce013a4e11ba75e87f5f8ce9eb>:0 
  at CKAN.CmdLine.Install.RunCommand (CKAN.KSP ksp, System.Object raw_options) [0x0020e] in <fce8ecce013a4e11ba75e87f5f8ce9eb>:0 
  at CKAN.CmdLine.MainClass.RunSimpleAction (CKAN.CmdLine.Options cmdline, CKAN.CmdLine.CommonOptions options, System.String[] args, CKAN.IUser user, CKAN.KSPManager manager) [0x00306] in <fce8ecce013a4e11ba75e87f5f8ce9eb>:0 
  at CKAN.CmdLine.MainClass.Main (System.String[] args) [0x001b5] in <fce8ecce013a4e11ba75e87f5f8ce9eb>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: CKAN.FileNotFoundKraken: Trying to install CivilianPopulation 2.0.12, but it's not downloaded or download is corrupted
  at CKAN.ModuleInstaller.Install (CKAN.CkanModule module, System.String filename) [0x00073] in <fce8ecce013a4e11ba75e87f5f8ce9eb>:0 
  at CKAN.ModuleInstaller.InstallList (System.Collections.Generic.ICollection`1[T] modules, CKAN.RelationshipResolverOptions options, CKAN.IDownloader downloader) [0x00194] in <fce8ecce013a4e11ba75e87f5f8ce9eb>:0 
  at CKAN.ModuleInstaller.InstallList (System.Collections.Generic.List`1[T] modules, CKAN.RelationshipResolverOptions options, CKAN.IDownloader downloader) [0x00028] in <fce8ecce013a4e11ba75e87f5f8ce9eb>:0 
  at CKAN.CmdLine.Install.RunCommand (CKAN.KSP ksp, System.Object raw_options) [0x0020e] in <fce8ecce013a4e11ba75e87f5f8ce9eb>:0 
  at CKAN.CmdLine.MainClass.RunSimpleAction (CKAN.CmdLine.Options cmdline, CKAN.CmdLine.CommonOptions options, System.String[] args, CKAN.IUser user, CKAN.KSPManager manager) [0x00306] in <fce8ecce013a4e11ba75e87f5f8ce9eb>:0 
  at CKAN.CmdLine.MainClass.Main (System.String[] args) [0x001b5] in <fce8ecce013a4e11ba75e87f5f8ce9eb>:0 
```

wget output:

```
wget https://github.com/rleroy/CivilianPopulation/releases/download/2.0.12/ksp-civilian-population-mod-2.0.12.zip
--2018-02-14 23:07:12--  https://github.com/rleroy/CivilianPopulation/releases/download/2.0.12/ksp-civilian-population-mod-2.0.12.zip
Herleiden van github.com (github.com)... 192.30.253.112, 192.30.253.113
Verbinding maken met github.com (github.com)|192.30.253.112|:443... verbonden.
HTTP-verzoek is verzonden; wachten op antwoord... 302 Found
Locatie: https://github-production-release-asset-2e65be.s3.amazonaws.com/78369832/d5efc812-fb11-11e7-895a-2f1b7fdbb7d8?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20180214%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20180214T220713Z&X-Amz-Expires=300&X-Amz-Signature=68c639f6b1599dcbcc809b649de4410a35cdeb1ec0acddd7586714bc620d7b09&X-Amz-SignedHeaders=host&actor_id=0&response-content-disposition=attachment%3B%20filename%3Dksp-civilian-population-mod-2.0.12.zip&response-content-type=application%2Foctet-stream [volgen...]
--2018-02-14 23:07:13--  https://github-production-release-asset-2e65be.s3.amazonaws.com/78369832/d5efc812-fb11-11e7-895a-2f1b7fdbb7d8?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20180214%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20180214T220713Z&X-Amz-Expires=300&X-Amz-Signature=68c639f6b1599dcbcc809b649de4410a35cdeb1ec0acddd7586714bc620d7b09&X-Amz-SignedHeaders=host&actor_id=0&response-content-disposition=attachment%3B%20filename%3Dksp-civilian-population-mod-2.0.12.zip&response-content-type=application%2Foctet-stream
Herleiden van github-production-release-asset-2e65be.s3.amazonaws.com (github-production-release-asset-2e65be.s3.amazonaws.com)... 52.216.18.56
Verbinding maken met github-production-release-asset-2e65be.s3.amazonaws.com (github-production-release-asset-2e65be.s3.amazonaws.com)|52.216.18.56|:443... verbonden.
HTTP-verzoek is verzonden; wachten op antwoord... 200 OK
Lengte: 14759554 (14M) [application/octet-stream]
Wordt opgeslagen als: ?ksp-civilian-population-mod-2.0.12.zip?

ksp-civilian-popula 100%[===================>]  14.08M  3.07MB/s    in 5.2s    

2018-02-14 23:07:19 (2.70 MB/s) - '?ksp-civilian-population-mod-2.0.12.zip?' opgeslagen [14759554/14759554]
```

--- a/Core/Net/NetFileCache.cs
+++ b/Core/Net/NetFileCache.cs
@@ -209,9 +209,9 @@ namespace CKAN
         }
 
         /// <summary>
-        /// Check whether a ZIP file is validation
+        /// Check whether a ZIP file is valid
         /// </summary>
-        /// <param name="filename">path to zip file to check</param>
+        /// <param name="filename">Path to zip file to check</param>
         /// <param name="invalidReason">Description of problem with the file</param>
         /// <returns>
         /// True if valid, false otherwise. See invalidReason param for explanation.
@@ -224,15 +224,27 @@ namespace CKAN
                 {
                     using (ZipFile zip = new ZipFile(filename))
                     {
-                        // Perform CRC check.
-                        if (zip.TestArchive(true))
+                        string zipErr = null;
+                        // Perform CRC and other checks
+                        if (zip.TestArchive(true, TestStrategy.FindFirstError,
+                            (TestStatus st, string msg) =>
+                            {
+                                // This delegate is called as TestArchive proceeds through its
+                                // steps, both routine and abnormal.
+                                // The second parameter is non-null if an error occurred.
+                                if (st != null && !st.EntryValid && !string.IsNullOrEmpty(msg))
+                                {
+                                    // Capture the error string so we can return it
+                                    zipErr = $"Error in step {st.Operation} for {st.Entry?.Name}: {msg}";
+                                }
+                            }))
                         {
                             invalidReason = "";
                             return true;
                         }
                         else
                         {
-                            invalidReason = "ZipFile.TestArchive(true) returned false";
+                            invalidReason = zipErr ?? "ZipFile.TestArchive(true) returned false";
                             return false;
                         }
                     }

